src/shogun/multiclass/ecoc/ECOCRandomDenseEncoder.cpp

```diff
@@ -67,7 +68,7 @@ SGMatrix<int32_t> CECOCRandomDenseEncoder::create_codebook(int32_t num_classes)
         {
             for (int32_t j=0; j < num_classes; ++j)
             {
-                float64_t randval = CMath::random(0.0, 1.0);
+                float64_t randval = random::random(0.0, 1.0, m_prng);
```

Comment:
this is my problem with these utility functions... again here you can just create the uniformdistr outside of this loop and sample it here..